nixos/modules/services/misc/paperless.nix

```diff
@@ -160,32 +163,30 @@ in
       description = lib.mdDoc "Web interface port.";
     };
 
-    # FIXME this should become an RFC42-style settings attr
-    extraConfig = mkOption {
-      type = types.attrs;
+    settings = mkOption {
+      type = lib.types.submodule {
+        freeformType = with lib.types; attrsOf (let
+          typeList = [ bool int str path package ];
+        in oneOf (typeList ++ [ (listOf (types.enum typeList)) (attrsOf (types.enum typeList)) ]));
```

Comment:
Took me a couple of tries to get this type correct and it kinda got a monster. If someone has an idea how to simplify this, please speak up. :)

Reply:
types.enum is definitely wrong here and breaks my own config...